scripts/qa_crowbarsetup.sh

```diff
@@ -3492,10 +3512,91 @@ function onadmin_run_cct()
     return 0
 }
 
+# Set the aliases for nodes.
+# This is usually needed before batch step, so batch can refer
+# to node aliases in the scenario file.
+function onadmin_setup_aliases()
+{
+    local nodesavailable=`get_all_discovered_nodes`
+    local i
+
+    if [ -n "$want_node_aliases" ] ; then
+        # aliases provided explicitely, assign them successively to the nodes
+        # example: want_node_aliases=controller=1,swift=2,kvm=2
+
+        for aliases in ${want_node_aliases//,/ } ; do
+
+            # split off the number => group
+            node_alias=${aliases%=*}
+            # split off the group => number
+            number=${aliases#*=}
+
+            i=1
+            for node in `printf  "%s\n" $nodesavailable | head -n$number`; do
+                if [[ $i -gt 1 ]]; then
+                    node_alias="$node_alias$i"
+                fi
+                set_node_alias $node $node_alias
+                nodesavailable=`printf "%s\n" $nodesavailable | grep -iv $node`
+                i=$((i+1))
+            done
+        done
+    else
+        # try to setup aliases automatically
+        if [[ $hacloud = 1 ]] ; then
+            # 1. HA
+            # use the logic from cluster_node_assignment and assign
+            #      dataN, serviceN, networkN aliases
+            # for nodes in clusternodesdata etc.
+
+            cluster_node_assignment
+
+            local i
+            for clustername in data network services ; do
+                eval "cluster=\$clusternodes$clustername"
+                i=1
+                for node in $cluster ; do
+                    set_node_alias $node "$clustername$i"
+                    i=$((i+1))
+                done
+            done
+            i=1
+            for node in $unclustered_nodes ; do
+                set_node_alias_and_role $node "compute$i" compute
+                i=$((i+1))
+            done
+        else
+            # 2. non-HA
+            # 1st node is controller by default (intended role is set by onadmin_allocate)
+            local controller=`get_all_discovered_nodes  | head -n1`
+            set_node_alias $controller "controller"
+            nodesavailable=`printf "%s\n" $nodesavailable | grep -iv $controller`
+
+            i=1
+            # storage nodes (cephN or swiftN) will exist based on deployceph/deployswift value
+            if [ -n "$deployceph" ] || [ -n "$deployswift" ] ; then
+                for node in `get_all_discovered_nodes | grep -v $controller | head -n2` ; do
+                    set_node_alias_and_role $node "storage$i" storage
+                    nodesavailable=`printf "%s\n" $nodesavailable | grep -iv $node`
+                    i=$((i+1))
+                done
+            fi
+
+            # Use computeN for the rest.
+            i=1
+            for node in $nodesavailable; do
+                set_node_alias_and_role $node "compute$i" compute
```

Comment:
(and again)

Alternatively, the function should be renamed to something else to make it clear it's not about just setting aliases.